modelexpress_common/src/providers/huggingface.rs

```diff
@@ -218,6 +221,33 @@ impl ModelProviderTrait for HuggingFaceProvider {
         Ok(())
     }
 
+    /// Get the full path to the latest model snapshot if it exists
+    /// Returns the path if found, or an error if not found
+    async fn get_model_path(&self, model_name: &str, cache_dir: PathBuf) -> Result<PathBuf> {
+        let normalized_name = model_name.replace("/", "--");
+        let path = cache_dir
+            .join(format!["models--{normalized_name}"])
+            .join("snapshots");
+
+        if !path.exists() {
+            anyhow::bail!("Model snapshots for '{model_name}' not found in cache");
+        }
+
+        let mut files: Vec<fs::DirEntry> = fs::read_dir(path)?.filter_map(Result::ok).collect();
+        files.sort_by_key(|e| {
+            e.metadata()
+                .and_then(|m| m.created().or_else(|_| m.modified()))
+                .unwrap_or(std::time::SystemTime::UNIX_EPOCH)
+        });
+        files.reverse(); // newest first
```

Comment:
A bit hacky way to figure out the latest snapshot (if multiple exists) since system time is not an ideal way to compare time. we probably want to come up with more robust approach (e.g. timestamping for every model download maybe?) but i think it deserves a separate discussion.